### PR TITLE
Enhance build process

### PR DESCRIPTION
**Pull request by jlec**
* Allow optimization flgas in CFLAGS to be overwritten
* Use Uppercase CC and CFLAGS
* Create all soname symlinks

Signed-off-by: Justin Lecher <jlec@gentoo.org>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,12 +2,13 @@
 
 # programs
 MAKE=make
-AR=ar
+AR?=ar
```
Comment (stevengj) [inline Makefile]: Again, why is this needed?   `AR=ar` can be overridden by `make AR=....` already.

Comment (jlec) [inline Makefile]: Yes, in case your environment has everything set, you just need to run ```make```. Especially when doing crosscompilation it is important not to call e.g. ```ar``` but the target platform specific version.
